fairroot/base/sim/FairIon.h

```diff
@@ -40,10 +40,10 @@ class FairIon : public TNamed
      ** If mass is not given, it will be set to a times the proton mass.
      **/
 
-    FairIon(const char* name, Int_t z, Int_t a, Int_t q, Double_t e = 0., Double_t m = 0.);
-    void SetParams(const char* name, Int_t z, Int_t a, Int_t q, Double_t e = 0., Double_t m = 0.)
+    FairIon(const std::string name, Int_t z, Int_t a, Int_t q, Double_t e = 0., Double_t m = 0.);
+    void SetParams(const std::string name, Int_t z, Int_t a, Int_t q, Double_t e = 0., Double_t m = 0.)
```

Comment:
Just leave the `const char*` here. Taking the `std::string` by-value just adds a useless copy. Unfortunately, `TNamed::SetName()` does not offer any suitable overload that allows modernization.